src/adapter.rs

```diff
@@ -55,8 +55,16 @@ pub trait BluetoothAdapter {
    fn set_id(&self, id: String)-> Result<(), Box<Error>>{
         Err(Box::from(NOT_SUPPORTED_ERROR))
      }
-    fn get_devices(&self)-> Result<Vec<BluetoothDevice>, Box<Error>>;
-    fn get_device(&self, address: String) -> Result<Option<BluetoothDevice>, Box<Error>>;
+    fn get_devices(&self)-> Result<Vec<BluetoothDevice>, Box<Error>>{
+        Err(Box::from(NOT_SUPPORTED_ERROR))   
+    }
+    fn get_device(&self, address: String) -> Result<Option<BluetoothDevice>, Box<Error>>{
```

Comment:
What is the motivation for these changes? I would expect all implementors to provide meaningful implementations for these methods instead.

Reply:
Same confusion here too. 
So what happened was, when we were asked to implement trait for EmptyAdapter in empty.rs, 
I found that it did not have methods like get_device, create_discovery_session. I should have added these functions in the implementation directly, instead of having default implementations here. My mistake, sorry